server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMailboxesCreationProcessor.java

```diff
@@ -134,6 +135,9 @@ private void createMailbox(MailboxCreationId mailboxCreationId, MailboxCreateReq
 
     private void ensureValidMailboxName(MailboxCreateRequest mailboxRequest, MailboxSession mailboxSession) {
         String name = mailboxRequest.getName();
+        if (name.length() >= MailboxConstants.DEFAULT_LIMIT_MAILBOX_NAME_SIZE) {
+            throw new MailboxNameException(String.format("The mailbox name length '%s' is over limitation: %s", name, MailboxConstants.DEFAULT_LIMIT_MAILBOX_NAME_SIZE));
+        }
```

Comment:
That's strange to handle it there. I think it's the mailboxManager responsibility to handle too long name and to give an exception when a too long name is provided. WDYT?